src/api/services/iuf/sessions.go

```diff
@@ -354,78 +354,82 @@ func (s iufService) ProcessOutput(session *iuf.Session, workflow *v1alpha1.Workf
-				}
-				jsonManifest, _ := json.Marshal(manifest)
-				s.logger.Infof("manifest: %s - %s", manifest["name"], manifest["version"])
-				if idx := slices.IndexFunc(activity.Products, func(product iuf.Product) bool { return product.Name == manifest["name"] }); idx == -1 {
-					// add product to activity object
-					activity.Products = append(activity.Products, iuf.Product{
-						Name:             fmt.Sprintf("%v", manifest["name"]),
-						Version:          fmt.Sprintf("%v", manifest["version"]),
-						Validated:        validated,
-						Manifest:         string(jsonManifest),
-						OriginalLocation: nodeStatus.Outputs.Parameters[1].Value.String(),
-					})
-					activity.OperationOutputs["stage_params"].(map[string]interface{})["process-media"].(map[string]interface{})["products"].(map[string]interface{})[fmt.Sprintf("%v", manifest["name"])] = make(map[string]interface{})
-					activity.OperationOutputs["stage_params"].(map[string]interface{})["process-media"].(map[string]interface{})["products"].(map[string]interface{})[fmt.Sprintf("%v", manifest["name"])].(map[string]interface{})["parent_directory"] = nodeStatus.Outputs.Parameters[1].Value.String()
-				}
-			}
-			session.Products = activity.Products
-			// update activity
-			configmap, err := s.iufObjectToConfigMapData(activity, activity.Name, LABEL_ACTIVITY)
+			err := s.processOutputOfProcessMedia(&activity, workflow)
 			if err != nil {
 				s.logger.Error(err)
 				return err
 			}
-
-			_, err = s.k8sRestClientSet.
-				CoreV1().
-				ConfigMaps(DEFAULT_NAMESPACE).
-				Update(
-					context.TODO(),
-					&configmap,
-					v1.UpdateOptions{},
-				)
+			session.Products = activity.Products
+			// update activity
+			_, err = s.updateActivity(activity)
 			if err != nil {
 				s.logger.Error(err)
 				return err
 			}
 			return nil
+		} else {
+			for _, task := range tasks {
+				operationName := task.TemplateRef.Name
+				nodeStatus := workflow.Status.Nodes.FindByDisplayName(task.Name)
+				s.logger.Infof("process output of: %s, %v", operationName, nodeStatus.Outputs)
+				s.updateActivityOperationOutputFromWorkflow(activity, *session, nodeStatus, operationName, "")
+			}
+			return nil
 		}
-		for _, task := range tasks {
-			operationName := task.TemplateRef.Name
-			nodeStatus := workflow.Status.Nodes.FindByDisplayName(task.Name)
-			s.logger.Infof("process output of: %s, %v", operationName, nodeStatus.Outputs)
-			s.updateActivityOperationOutputFromWorkflow(activity, *session, nodeStatus, operationName, "")
-		}
-		return nil
 	default:
 		return fmt.Errorf("stage_type: %s is not supported", workflow.Labels["stage_type"])
 	}
 
 }
 
+func (s iufService) processOutputOfProcessMedia(activity *iuf.Activity, workflow *v1alpha1.Workflow) error {
+	nodesWithOutputs := workflow.Status.Nodes.Filter(func(nodeStatus v1alpha1.NodeStatus) bool {
+		return nodeStatus.Outputs.HasOutputs() && len(nodeStatus.Outputs.Parameters) == 2
+	})
+	if len(nodesWithOutputs) == 0 {
+		return nil
```

Comment:
yep

Reply:
Yes. For example, this would happen when the media directory does not contain any `*.tar.gz` release dist files. The result would be that the subsequent stages would be zero-trip loops. This is harmless.